### PR TITLE
group 2: updated book & user scripts

### PR DESCRIPTION
**Pull request by janavatter**

### CODE REVIEW

File: library/model/book.py [hunk 1]
```diff
@@ -35,6 +56,8 @@ def __init__(self, title, authors, publisher, pub_date, genres, pages, isbn, typ
         self.duration = duration
         self.existing_items = existing_items
         self.borrowed_items = borrowed_items
+        self._fee_by_booktype = BookType
```
Comment (maxftl) [inline library/model/book.py:59]: _fee_by_booktype should be a dict

File: library/model/book.py [hunk 2]
```diff
@@ -35,6 +56,8 @@ def __init__(self, title, authors, publisher, pub_date, genres, pages, isbn, typ
...
+        self._duration_by_booktype = BookType
```
Comment (maxftl) [inline library/model/book.py:60]: should be a dictionary

Comment (sajid3900) [inline library/model/book.py:60]: I guess this is not required

File: library/model/book.py [hunk 3]
```diff
@@ -64,24 +87,35 @@ def can_borrow(self) -> bool:
...
-        if self._book_type == "Paper":
-            return 5
-        elif self._book_type == "Electronic":
-            return 2
-        elif self._book_type == "Audio":
-            return 2
-        else:
-            raise AttributeError("No such book type...")
+        try:
+            print(self._book_type, self.fee_by_book_type.get(self._book_type, 0), self.fee_by_book_type)
```
Comment (sajid3900) [inline library/model/book.py:106]: why print?

File: library/model/book.py [hunk 4]
```diff
@@ -23,6 +31,19 @@ class Book:
...
+    fee_by_book_type: dict[BookType, int] = {
+        BookType.PAPER: 5,
+        BookType.ELECTRONIC: 2,
+        BookType.AUDIO: 2
+    }
+
+    duration_by_book_type: dict[BookType, int] = {
+        BookType.PAPER: 180,
+        BookType.ELECTRONIC: 300,
+        BookType.AUDIO: 1
```
Comment (sajid3900) [inline library/model/book.py:43]: Audio 1? You sure?

File: library/model/book.py [hunk 5]
```diff
@@ -64,24 +87,35 @@ def can_borrow(self) -> bool:
...
-        if self._book_type == "Paper":
-            return self.pages * 3 * 60
-        elif self._book_type == "Electronic":
-            return self.pages * 5 * 60
-        elif self._book_type == "Audio":
-            return self.duration
-        else:
-            raise AttributeError("No such book type...")
+        try:
+            return self._duration_by_booktype.get(self._book_type, 0)
```
Comment (sajid3900) [inline library/model/book.py:91]: The approx duration is self.pages *  time per page (in mins). This is incorrect.

File: library/model/book.py [hunk 6]
```diff
@@ -64,24 +87,35 @@ def can_borrow(self) -> bool:
...
-            return 2
-        else:
-            raise AttributeError("No such book type...")
+        try:
+            print(self._book_type, self.fee_by_book_type.get(self._book_type, 0), self.fee_by_book_type)
+            return self.fee_by_book_type.get(self._book_type, 0)
+        except AttributeError as e:
+            logging.error("No such book type...")
+        return 0
+        #if self._book_type == "Paper":
```
Comment (sajid3900) [inline library/model/book.py:111]: remove comments

File: library/model/book.py [hunk 7]
```diff
@@ -64,24 +87,35 @@ def can_borrow(self) -> bool:
...
-        elif self._book_type == "Audio":
-            return self.duration
-        else:
-            raise AttributeError("No such book type...")
+        try:
+            return self._duration_by_booktype.get(self._book_type, 0)
+        except AttributeError as e:
+            logging.error("No such book type...")
+        return 0
+        #if self._book_type == "Paper":
```
Comment (sajid3900) [inline library/model/book.py:95]: remove comments

**Review by sajid3900 — CHANGES_REQUESTED** [review]
Changes required.

**Pull request closed without merging**